SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::GenerateDevAuthToken()
+{
+	if (!bIsGenerating)
+	{
+		bIsGenerating = true;
+		AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this]
+			{
+				AsyncTask(ENamedThreads::GameThread, [this]()
+					{
+						this->ShowTaskStartedNotification(TEXT("Generating Developer Authentication Token"));
+					});
+				FString DevAuthToken;
+				USpatialGDKSettings* GDKRuntimeSettings = GetMutableDefault<USpatialGDKSettings>();
+				bool bSuccess = SpatialCommandUtils::GenerateDevAuthToken(GDKRuntimeSettings->IsRunningInChina(), DevAuthToken);
+
+				USpatialGDKEditorSettings* GDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+				GDKEditorSettings->DevelopmentAuthenticationToken = DevAuthToken;
+				GDKEditorSettings->SaveConfig();
+				GDKEditorSettings->SetRuntimeDevelopmentAuthenticationToken();
+
+				// Ensure we enable bUseDevelopmentAuthenticationFlow when using cloud deployment flow.
+				GDKRuntimeSettings->bUseDevelopmentAuthenticationFlow = true;
+				GDKRuntimeSettings->DevelopmentAuthenticationToken = DevAuthToken;
+
+				if (bSuccess)
+				{
+					AsyncTask(ENamedThreads::GameThread, [this]()
+						{
+							this->ShowTaskEndedNotification(TEXT("Developer Authentication Token Updated"), SNotificationItem::CS_Success);
+						});
+				}
+				else
+				{
+					UE_LOG(LogSpatialGDKDevAuthTokenGenerator, Error, TEXT("Failed to generate a development authentication token."));
+					AsyncTask(ENamedThreads::GameThread, [this]()
+						{
+							this->ShowTaskEndedNotification(TEXT("Developer Authentication Token Updated"), SNotificationItem::CS_Fail);
+						});
+
+				}
+
+				bIsGenerating = false;
```

Comment:
It is better to execute the above logic in the same thread with the caller thread, or else it may cause some unexpected issues.